### ISSUE TITLE
Console removed - what to change?

### ISSUE THREAD

**Issue by mauricioszabo** [JunoLab/atom-ink#219]
Today I found out that the old Console was removed on the new Ink version. How do I change the API? Is there any information about it? It broke completely my Chlorine plug-in, as it depends on a Console window currently opened...

The problem in this issue is that there's no documentation on how to migrate to `InkTerminal` and also no information on the CHANGELOG, so it's a huge breaking change with no visibility whatsoever...

**Comment by pfitzseb** [comment]
> The problem in this issue is that there's no documentation on how to migrate to InkTerminal and also no information on the CHANGELOG, so it's a huge breaking change with no visibility whatsoever...

Fair point. It's probably not really a good idea to treat `ink` as a library at this point, unfortunately.

There's no easy way to migrate, I think, but the easiest thing would be to directly start a REPL process and attach that to the `InkTerminal`. If you have only have `stdin`/`stdout`/`stderr` streams then things are probably more complicated, and I'm honestly not sure how that would end up working.

If you want to continue using the old console then I suppose we could re-add that, but I'd recommend just copying the relevant code ([this](https://github.com/JunoLab/atom-ink/tree/v0.10.12/lib/console), [this](https://github.com/JunoLab/atom-ink/blob/v0.10.12/lib/util/pane-item.js), [this](https://github.com/JunoLab/atom-ink/blob/v0.10.12/lib/util/ansitohtml.js), and [this](https://github.com/JunoLab/atom-ink/blob/v0.10.12/lib/util/etch.js)) over to your repo and drop the dependency on ink.

Let me know what you want to go with and I'd be happy to help.

**Comment by mauricioszabo** [comment]
Well, I need the old functionality because I don't start any external process, so I need the stdout and stderr.

I was also using the console as a drop-in for ink results, so I need to render HTML (one thing that the new implementation don't support either). I'll probably just end up copying the code over my repo...

**Comment by pfitzseb** [comment]
Alright. Sorry for the inconvenience then, but we can't really keep maintaining the old console and removed it to improve loading times.

**Comment by aviatesk** [comment]
@mauricioszabo 
I missed to check this issue, and sorry for the inconvenience ... (I'm the one who removed the legacy console support)
But atst I think it might be better for your to just borrow REPL implementation into your repo as @pfitzseb pointed out, since the other Ink components (and the package interaction in general) are actually hurting your package loading/activation time (maybe A LOT).
